content/docs/forwarding-refs.md

```diff
@@ -1,76 +1,73 @@
 
-## Forwarding refs to DOM components {#forwarding-refs-to-dom-components}
+## Transfert de refs vers des composants du DOM {#forwarding-refs-to-dom-components}
 
-Consider a `FancyButton` component that renders the native `button` DOM element:
+Prenons un composant `FancyButton` qui affiche l'élément DOM natif `button` :
 `embed:forwarding-refs/fancy-button-simple.js`
 
-React components hide their implementation details, including their rendered output. Other components using `FancyButton` **usually will not need to** [obtain a ref](/docs/refs-and-the-dom.html) to the inner `button` DOM element. This is good because it prevents components from relying on each other's DOM structure too much.
+Les composants React masquent leurs détails d'implémentation, y compris leur sortie du moteur de rendu. *(Dans la suite de cet article, pour des raisons de concision, nous emploierons le terme générique anglais renderer sans italiques, NdT)*
+Les autres composants utilisant `FancyButton` **n'auront généralement pas besoin** [d'obtenir une ref](/docs/refs-and-the-dom.html) à l'intérieur de l'élément du DOM `button`. C'est une bonne chose, car ça empêche les composants de trop s'appuyer sur la structure du DOM des uns et des autres.
 
-Although such encapsulation is desirable for application-level components like `FeedStory` or `Comment`, it can be inconvenient for highly reusable "leaf" components like `FancyButton` or `MyTextInput`. These components tend to be used throughout the application in a similar manner as a regular DOM `button` and `input`, and accessing their DOM nodes may be unavoidable for managing focus, selection, or animations.
+Bien qu'une telle encapsulation soit souhaitable pour les composants au niveau de l'application tels que `FeedStory` ou `Comment`, elle peut être gênante pour les composants hautement réutilisables, tels que `FancyButton` ou `MyTextInput`. Ces composants ont tendance à être utilisés dans l’application de la même manière qu’un `bouton` et un `input` « normaux », et l’accès à leurs nœuds DOM peut être inévitable pour la gestion du focus, de la sélection ou des animations.
 
-**Ref forwarding is an opt-in feature that lets some components take a `ref` they receive, and pass it further down (in other words, "forward" it) to a child.**
-
-In the example below, `FancyButton` uses `React.forwardRef` to obtain the `ref` passed to it, and then forward it to the DOM `button` that it renders:
+**Le transfert de Ref est une fonctionnalité d’inscription qui permet à certains composants de prendre une `ref` qu’ils reçoivent et de les passer plus loin (en d’autres termes, « les transférer ») à un enfant.**
```

Comment:
```suggestion
**Le transfert de ref est une fonctionnalité optionnelle qui permet à certains composants de prendre une `ref` qu’ils reçoivent et de la passer plus bas dans l’arbre (en d’autres termes, la « transférer ») à un composant enfant.**
```

Ici, “opt-in” indique qu'elle n'est pas active par défaut, il nous faut recourir à du code explicite pour en bénéficier. Rien à voir avec l'inscription…